### ISSUE TITLE
`@stylistic/value-list-comma-space-before` reports the whitespace in front of a comma that opens the value and never fixes it

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#166]
Split out of [#134](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/134), which moves the same boundary in the two `*-after` rules. This side needs more than a boundary moved, and is left standing there.

### What minimal example or steps are needed to reproduce the bug?

```css
a { prop: ,0; }
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "plugins": [
    "@stylistic/stylelint-plugin"
  ],
  "rules": {
    "@stylistic/value-list-comma-space-before": "never"
  }
}
```

### How did you run Stylelint?

```shell
stylelint "**/*.css" --config myconfig.json --fix
```

### Which Stylelint-related dependencies are you using?

```json
{
  "@stylistic/stylelint-plugin": "5.3.0",
  "stylelint": "17.14.1"
}
```

### What did you expect to happen?

```css
a { prop:,0; }
```

### What actually happened?

The problem is reported and never fixed, however many times `--fix` runs:

```
1:11  Unexpected whitespace before "," (@stylistic/value-list-comma-space-before)
```

The warning does survive the run, so nothing is lost silently — it simply never clears. `never` and `never-single-line` both do it, and a custom property behaves the same:

```css
/* reported for ever */   a { prop: ,0; }
/* reported for ever */   a { prop:  ,0; }
/* reported for ever */   a { --a: ,0; }
```

The `always` options have nothing to report here: the whitespace they ask for is already in place, since a declaration is spelled with a space after its colon.

### Do you have a proposal to fix the bug?

The rule decides what it may touch by

```js
isFixable: (declNode, index, declString) => index > declarationValueIndex(declNode) && !endsWithInlineComment(declString.slice(0, index)),
```

and widening that boundary to `>=`, as #134 does for `value-list-comma-space-after` and `value-list-comma-newline-after`, is wrong here. For a comma opening the value, the whitespace in front of it is not part of the value at all: it is `decl.raws.between`, the text standing between the colon and the value. This rule writes through `setDeclarationValue` and cannot reach it, so with `>=` it would silence its own warning and change nothing:

```
without --fix   1:11
with --fix      nothing, and the code is untouched
relinting it    1:11
```

which is a fresh instance of [#126](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/126).

Curing it means teaching the rule to write `decl.raws.between` when the comma it is fixing is the value's first character: take the whitespace off the end of `between` under `never`, and put a single space there under `always` — the same text `declarationValueIndex` counts, so the indexes the checker reports stay where they are as long as the write happens after the reporting.

Worth deciding along with it: `a { prop:,0; }` is what `never` asks for, and it is a declaration with no whitespace after its colon, which `declaration-colon-space-after` has an opinion about. Two rules writing the same run of whitespace from opposite ends is the sort of thing that should be checked to converge before it ships.
